pkg/api/validation/validation.go

```diff
@@ -2562,6 +2572,17 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(specPath.Child("activeDeadlineSeconds"), newPod.Spec.ActiveDeadlineSeconds, "must not update from a positive integer to nil value"))
 	}
 
+	// validate updated spec.PriorityClassName.
+	if newPod.Spec.PriorityClassName != oldPod.Spec.PriorityClassName && len(newPod.Spec.PriorityClassName) > 0 {
```

Comment:
Yes, this is allowed. An empty priority class name is resolved to the "default" priority. One of the priority classes can be annotated as default and its value will be used as default priority.